fabric-rendering-fluids-v1/src/main/java/net/fabricmc/fabric/mixin/client/rendering/fluid/MixinFluidRenderer.java

```diff
@@ -42,16 +46,41 @@
 	private Sprite[] lavaSprites;
 	@Shadow
 	private Sprite[] waterSprites;
+	@Shadow
+	private Sprite waterOverlaySprite;
 
 	private final ThreadLocal<FluidRendererHookContainer> fabric_renderHandler = ThreadLocal.withInitial(FluidRendererHookContainer::new);
+	private final ThreadLocal<Boolean> fabric_customRendering = ThreadLocal.withInitial(() -> false);
+	private final ThreadLocal<Block> fabric_neighborBlock = new ThreadLocal<>();
 
 	@Inject(at = @At("RETURN"), method = "onResourceReload")
 	public void onResourceReloadReturn(CallbackInfo info) {
-		FluidRenderHandlerRegistryImpl.INSTANCE.onFluidRendererReload(waterSprites, lavaSprites);
+		FluidRenderer self = FluidRenderer.class.cast(this);
```

Comment:
```suggestion
		FluidRenderer self = (FluidRenderer) (Object) this;
```
More idiomatic, at least for Fabric API

Reply:
I think this is a smart way to avoid the warning. :smile: